ext/lists.go

```diff
@@ -65,6 +90,26 @@ func (listsLib) CompileOptions() []cel.EnvOption {
 				}),
 			),
 		),
+		cel.Function("flatten",
+			cel.MemberOverload("list_flatten",
+				[]*cel.Type{listDyn}, listDyn,
+				cel.UnaryBinding(func(arg ref.Val) ref.Val {
+					list := arg.(traits.Lister)
+					flatList := flattenHelper(list, false)
+					return types.DefaultTypeAdapter.NativeToValue(flatList)
+				}),
+			),
+		),
+		cel.Function("flattenDeep",
```

Comment:
Instead of `flattenDeep`, can we consider adding an overload of `flatten()` that takes an explicit argument indicating how many levels deep to flatten, e.g. `list_flatten_int`? This is safer than unbounded recursion. I'd be open to having `-1` as the depth signal unbounded recursion, but we can at least make it really clear what the compute costs could be for flattening.